### ISSUE TITLE
Question about isomorphic/universal example

### ISSUE THREAD

**Issue by franvarney** [jedireza/hapi-react-views#26]
I'm pretty new to React and having a hard time understanding how multiple views (pages) are rendered when using this module to set up an isomorphic application. 

I see that client.js calls render on app.jsx, but how does this work when you have more than just app.jsx? I created two separate views, but it looks like the second page is trying to render with app.jsx. How do I set this up to accept multiple views? If this isn't what I should be doing, can you tell me how I should be going about this?

**Comment by jedireza** [comment]
Thanks for opening an issue. The remount example demonstrating the ability to do universal (isomorphic) apps with `hapi-react-views` is super simple and leaves a lot to be desired.

[Aqua](https://github.com/jedireza/aqua/) uses `hapi-react-views` but doesn't use a universal approach. It's like a handful of single page apps that make up a larger app. It's not universal on purpose. Someone once asked ["How to make aqua isomorphic?"](https://github.com/jedireza/aqua/issues/39) and in that thread I talk about the different considerations we need to keep in mind to achieve universal web app nirvana. Some points I'd like to highlight form that thread:

 - Choosing an isomorphic Flux framework is a big decision that'll probably come with a little vendor lock-in.
 - Making your app isomorphic will touch your stack from top to bottom.
 - Routing on the client and server need to line up. __<-- this is an important part of your original question__
 - Flux on the server get's tricky because you can't use singletons and must isolate each request and you have to make actions asynchronous so you know when it's time to render the view.
 - You'll also need to have a data access layer done in a way so that on the server it accesses resources directly but on the client those same calls are proxied via ajax requests [...]

The React/Flux landscape has been changing a lot and I think these new things will change the way I approach universal web apps in the future.

 - [React Router](https://github.com/rackt/react-router) is approaching a v1 release and changes a lot from the latest stable release.
 - [Redux](https://github.com/rackt/redux) seems to be changing/simplifying the ideas of Flux and gaining lots of popularity.
 - [Relay](https://facebook.github.io/relay/) along with GraphQL also look promising for universal web apps development.

I haven't been keeping up with the React/Flux/universal-app landscape for a while and will need to dive back in. I'm waiting for `react-router` to release a v1 before approaching the next big Aqua update.

:open_mouth:  Looking at how much I described... it's a lot to take in. I hope this is helpful, even if only by shedding light on things.

**Comment by jedireza** [comment]
Closing this now. I hope the details helped. What did you end up going with?

**Comment by franvarney** [comment]
Thanks for your response @jedireza. Should have responded sooner but time got away from me! 

At the time I posted that, I ended up just doing a sort of basic React app. Figured I would focus on just learning React itself. Currently, I am working my own starter template using Hapi+React/Redux. There's a bit of a learning curve, but it's starting to make sense (slowly)! I will definitely look at some of your repos for reference. :)
